src/rabbit_delayed_message.erl

```diff
@@ -185,18 +187,7 @@ internal_delay_message(CurrTimer, Exchange, Delivery, Delay) ->
                        make_delay(DelayTS, Exchange, Delivery)),
     case erlang:read_timer(CurrTimer) of
         false ->
-            %% last timer expired, we set a new timer for
-            %% the next message to be delivered
-            case mnesia:dirty_first(?INDEX_TABLE_NAME) of
-                %% destructuring to prevent matching '$end_of_table'
-                #delay_key{timestamp = FirstTS} = Key
-                  when FirstTS < DelayTS ->
-                    %% there are messages that expired and need to be delivered
-                    {ok, start_timer(FirstTS - Now, Key)};
-                _ ->
-                    %% empty table or DelayTS <= FirstTS
-                    {ok, start_timer(Delay, make_key(DelayTS, Exchange))}
-            end;
+            {ok, CurrTimer};
```

Comment:
I filed https://github.com/rabbitmq/rabbitmq-delayed-message-exchange/pull/53 to explore this approach.  When we track the unset timer explicitly, we no longer need the idle timer, so the change looks quite different from what's proposed in this PR.